examples/sep24/sep24.ts

```diff
@@ -0,0 +1,189 @@
+import axios from "axios";
+import readline from "readline";
+
+import {
+  walletSdk,
+  Anchor,
+  SigningKeypair,
+  Types,
+  IssuedAssetId,
+} from "@stellar/typescript-wallet-sdk";
+import { Memo, MemoText } from "stellar-sdk";
+
+const wallet = walletSdk.Wallet.TestNet();
+const stellar = wallet.stellar();
+const anchor = wallet.anchor({ homeDomain: "testanchor.stellar.org" });
+const account = stellar.account();
+
+let kp: SigningKeypair;
+
+const asset = new IssuedAssetId(
+  "USDC",
+  "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5",
+);
+
+const runSep24 = async () => {
+  await createAccount();
+  await runDeposit(anchor, kp);
+  await runDepositWatcher(anchor);
+
+  while (!depositDone) {
```

Comment:
Do we not need to do it for withdrawal too?

Reply:
Also, I think it should be inside runDepositWatcher

Reply:
> Do we not need to do it for withdrawal too?

no the watcher keeps watching until stop() is called, which we only call when the status is returned as completed

> Also, I think it should be inside runDepositWatcher

similar the watcher for the deposit will also keep running until its stop() is called. The only reason there's a while loop is so we can listen for the deposit watcher to finish before asking to start the withdrawal flow. Imo it's a separate idea, since it's just there to connect the deposit to the withdrawal (and not needed unless you're running code like this)

Reply:
> no the watcher keeps watching until stop() is called, which we only call when the status is returned as completed

Won't the main thread die if we don't do the while loop after withdrawal flow?

Reply:
> Won't the main thread die if we don't do the while loop after withdrawal flow?

no the watcher will keep running until `stop` is called. See here: https://github.com/stellar/typescript-wallet-sdk/blob/8a7d22f432b23ad62b743486509d5394e548d155/src/walletSdk/Watcher/index.ts#L195-L205

once the call to the anchor is made, it waits for timeout length and then calls again it (and so on)